go-controller/pkg/ovn/secondary_layer2_network_controller_test.go

```diff
@@ -0,0 +1,208 @@
+package ovn
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"github.com/urfave/cli/v2"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	nadapi "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing"
+	libovsdbtest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/libovsdb"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+var _ = XDescribe("OVN Multi-Homed pod operations for layer2 network", func() {
+	var (
+		app       *cli.App
+		fakeOvn   *FakeOVN
+		initialDB libovsdbtest.TestSetup
+	)
+
+	BeforeEach(func() {
+		Expect(config.PrepareTestConfig()).To(Succeed()) // reset defaults
+
+		app = cli.NewApp()
+		app.Name = "test"
+		app.Flags = config.Flags
+
+		fakeOvn = NewFakeOVN(true)
+		initialDB = libovsdbtest.TestSetup{
+			NBData: []libovsdbtest.TestData{},
+		}
+
```

Comment:
should we have an IC test as well?

Reply:
Unfortunately, that will not do: the layer2 secondary network controller does **not** allocate the pod annotation when IC is used - the cluster manager does.

I could skip that check and tinker a bit with the expectations though ... Let me try.

Reply:
Hm, there's still some things needed - seems I need to skip the workload logical switch port creation (that requires the cluster manager involvement in allocating the OVN annotations).

I'll do it in a separate commit, and we can discuss if it is worth it.

Reply:
Done; let me know what you think @trozet .

Reply:
as we talked about probably better to put the annotation on the pod

Reply:
Done.